x-pack/plugins/infra/public/alerting/common/components/metrics_alert_dropdown.tsx

```diff
@@ -17,14 +17,17 @@ import { FormattedMessage } from '@kbn/i18n-react';
 import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
 import { PrefilledInventoryAlertFlyout } from '../../inventory/components/alert_flyout';
 import { PrefilledThresholdAlertFlyout } from '../../metric_threshold/components/alert_flyout';
-import { useRulesLink } from '../../../../../observability/public';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+
 type VisibleFlyoutType = 'inventory' | 'threshold' | null;
 
 export const MetricsAlertDropdown = () => {
   const [popoverOpen, setPopoverOpen] = useState(false);
   const [visibleFlyoutType, setVisibleFlyoutType] = useState<VisibleFlyoutType>(null);
   const uiCapabilities = useKibana().services.application?.capabilities;
-
+  const {
+    services: { observability },
+  } = useKibanaContextForPlugin();
```

Comment:
I'm in the process of deprecating this hook. Can you use [`useKibana`](https://github.com/elastic/kibana/blob/6693ef371f887eca639b09c4c9b15701b4ebabd4/src/plugins/kibana_react/README.md) instead?

The page should be wrapped already with the relevant Provider so it should work straightaway

Reply:
@afgomez I tried to use `useKibana`, but I was getting a typescript error:

`Property 'observability' does not exist on type 'Partial<CoreStart>'.` that's why I used `useKibanaContextForPlugin` instead. How do I overcome this Typescript error?

Reply:
Ok I refactored to this and works fine, so I'll get rid of `useKibanaContextForPlugin`

```  
const {
    services: { observability },
  } = useKibana<InfraClientStartDeps>();
```